pandas/core/indexes/base.py

```diff
@@ -7144,7 +7144,10 @@ def maybe_sequence_to_range(sequence) -> Any | range:
         return sequence
     if len(sequence) == 0:
         return range(0)
-    np_sequence = np.asarray(sequence, dtype=np.int64)
+    try:
+        np_sequence = np.asarray(sequence, dtype=np.int64)
+    except OverflowError:
```

Comment:
Why is the OverflowError required in this case? Have we added that elsewhere to support the RangeIndex return?

Reply:
This line will fail if the sequence contains an int over the max `int64` value e.g. `test_loc_uint64`.

This is the only pathway that can convert an arbitrary sequence to a RangeIndex return